src/test/java/bio/terra/service/duos/DuosClientTest.java

```diff
@@ -0,0 +1,112 @@
+package bio.terra.service.duos;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+import bio.terra.app.configuration.DuosConfiguration;
+import bio.terra.common.category.Unit;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.duos.exception.DuosDatasetBadRequestException;
+import bio.terra.service.duos.exception.DuosDatasetNotFoundException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@RunWith(SpringRunner.class)
+@ActiveProfiles({"google", "unittest"})
+@Category(Unit.class)
+public class DuosClientTest {
+
+  @MockBean private DuosConfiguration duosConfiguration;
+  @Mock private RestTemplate restTemplate;
+
+  private static final String BASE_PATH = "https://consent.dsde-dev.broadinstitute.org";
+  private static final String DUOS_ID = "DUOS-123456";
+  private static final Integer DUOS_DATASET_ID = 1;
+
+  private DuosClient duosClient;
+  private AuthenticatedUserRequest userReq;
+
+  @Before
+  public void setup() {
+    duosClient = new DuosClient(duosConfiguration, restTemplate);
+
+    when(duosConfiguration.getBasePath()).thenReturn(BASE_PATH);
+
+    userReq =
+        AuthenticatedUserRequest.builder()
+            .setSubjectId("DuosClientTest")
+            .setEmail("DuosClientTest@unit.com")
+            .setToken("token")
```

Comment:
This field is used, but no test asserts that it's being used property. Maybe it's too hard to test this with the way it's used?

Reply:
I had to play with this a bit, but ended up with a custom `ArgumentMatcher` to verify that the `HttpEntity` had the token set as its sole means of authentication.  This may be overkill in other similar tests but as we're apt to call the consent API as both an authenticated user and a TDR SA, I think it's worth it here.